nova/compute/api.py

```diff
@@ -4185,6 +4185,20 @@ def resize(self, context, instance, flavor_id=None, clean_shutdown=True,
         host_name can be set in the cold migration case only.
         """
 
+        # get all volumes attached to the instance and check if they are in a
+        # valid state for resize
+        bdms = objects.BlockDeviceMappingList.get_by_instance_uuid(
+                context, instance.uuid)
```

Comment:
Is the context already mapped to the right cell? If not, it would not find the BDMs and the check would do nothing.

I mean, it looks like it found the instance before ... but idk how it works.